docs/configuration.rst

```diff
@@ -42,9 +42,129 @@ File: config.yml
+
+sync_permissions:
+#################
+
+This section defines how to synchronize permissions between services when they share resources.
+This is used only for the synchronization of permissions between Magpie services/resources.
+The ``sync_permissions`` are defined first by a list of Magpie services and their associated resources,
+defined in the ``services`` section below. The mappings defining how the resources should be synchronized are described
+in the other section ``permissions_mapping``.
+
+services:
+_________
+
+This section contains the different resources that can be synchronized, ordered by service. The services found in
+this section of the config should also exist in Magpie.
+
+For more details on available services on Magpie, refer to these pages :
+
+- `Magpie available services <https://pavics-magpie.readthedocs.io/en/latest/services.html#available-services>`_
+- `Magpie services API <https://pavics-magpie.readthedocs.io/en/latest/autoapi/magpie/services/index.html>`_
+
+Each service defines a list of resource keys, which are custom names that define a resource path.
+They should correspond to the names used in the ``permissions_mapping`` section below.
+Each resource path contains the list of its segments, with their corresponding name and type.
+
+The name of a segment can either be a string name, a variable or a ``MULTI_TOKEN`` (``**``).
+
+Variables are indicated by a name written between braces (ex.: ``{variable_name}``) and represent a single
+segment name. A variable can be reused across different resource paths if they have
+a matching segment name. A resource path can use any number of different variables, but
+each variable can only be used one time per resource path.
+They are useful to indicate the corresponding location of the resource segment in a mapped permission.
+Note that all variables found in a target resource path should also be included in the source resource path.
+
+``MULTI_TOKEN`` represent any number (0 or more) of names that fit with the corresponding type.
+Also, the ``MULTI_TOKEN`` can only be used one time in each list of path segments. This is to avoid ambiguous
+cases that would result with using multiple ``MULTI_TOKEN``, since multiple ways of matching the resource path
+would then be possible. For example with a tokenized path ``**/**`` and an input resource ``seg1/seg2/seg3``,
+multiple choices of matching are possible.
+We could match ``seg1/seg2`` with the first token, and ``seg3`` with the second token,
+we could also match ``seg1`` with the first token, and ``seg2/seg3`` with the second token, etc.
+
+The variables and tokens are useful to know the type of any segments that doesn't have a fixed name.
+
+permissions_mapping:
+____________________
+
+This section defines an array of permissions mapping between services.
+Each item found in the ``permissions_mapping`` uses the following format :
+
+    ``resource_key1 : <permissions1> <mapping> resource_key2 : <permissions2>``
```

Comment:
I think we should use a `.. code-block:: yaml` here and add the quotes explicitly as in the real config. 
Otherwise, the `:` could be misinterpreted by in the config if the reader writes a definition literally like above.